apps/total-typescript/src/templates/book-chapter-template.tsx

```diff
@@ -154,21 +172,24 @@ const BookChapterTemplate = ({
                     )
                   },
                   h2: (props: any) => {
+                    const resourceId = resources?.find(
+                      (r) => r.slug === props.id,
+                    )?._id
```

Comment:
we find the Sanity `resourceId` on the h2 by checking the resources array in the current chapter and doing a find on slug